pkgs/applications/video/screenkey/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, distutils_extra
+, setuptools-git
+, intltool
+, pygtk
+, libX11
+, libXtst
+}:
+buildPythonApplication rec {
+  name = "screenkey-0.9";
+
+  src = fetchFromGitHub {
+    owner = "wavexx";
+    repo = "screenkey";
+    rev = name;
+    sha256 = "14g7fiv9n7m03djwz1pp5034pffi87ssvss9bc1q8vq0ksn23vrw";
+  };
+
+  nativeBuildInputs = [
+    distutils_extra
+    setuptools-git
+    intltool
+  ];
+
+  propagatedBuildInputs = [
+    pygtk
+  ];
+
+  makeWrapperArgs = [
+    "--set LD_LIBRARY_PATH ${libX11}/lib:${libXtst}/lib"
```

Comment:
Have you considered patching instead?

Reply:
Do you mean `patchelf`? If so, there is no ELF to patch.

https://github.com/wavexx/screenkey/blob/1b9d6c2dc1e4e9c39d85d756796bc8f54febeb97/Screenkey/xlib.py#L9

Reply:
See for example here https://github.com/NixOS/nixpkgs/blob/92513b8a41fc4d1d7863c803be9aa1f3746cef42/pkgs/tools/admin/salt/fix-libcrypto-loading.patch#L10

Reply:
You would create `paths.patch`

```patch
--- a/Screenkey/xlib.py
+++ b/Screenkey/xlib.py
@@ -6,7 +6,7 @@
 from ctypes import *
 
 ## base X11
-libX11 = CDLL('libX11.so.6')
+libX11 = CDLL('@libX11@/lib/libX11.so.6')
 
 # types
 Atom = c_ulong
@@ -278,7 +278,7 @@
 
 
 ## record extensions
-libXtst = CDLL('libXtst.so.6')
+libXtst = CDLL('@libXtst@/lib/libXtst.so.6')
 
 # types
 XPointer = String

```

Then you would add 

```nix
{
patches = [
  (substituteAll {
    src = ./paths.patch;
    inherit libX11 libXtst;
  })
];
}
```

Reply:
`.so` can be replaced with `${stdenv.hostPlatform.extensions.sharedLibrary}`